### PR TITLE
Fix/font name display

### LINKED ISSUE #105: [Bug] Font family name not displayed in font drop area when Google font is loaded.
### Current Behavior

Currently the state of the font file dropzone component does not update to show the google font name. When a google font is loaded it is the "current" font and its family name should be displayed in the font dropzone box to avoid confusion.

The dropzone updates the font family name just fine for files.

This is not a breaking bug, and more of a cosmetic improvement but it will improve useability.

### Relevant log output

_No response_

### Expected Behavior

- When no fonts have been loaded: Drop area reads "Drop a font".
- When a google font is loaded, drop area updates the font name text to the google font family name.
- When a font is dropped/loaded form file, drop area updates the font name to the file's font name.

### Changes Needed

Send FontUploader the google font family name (or have FontUploader track its state in App), such that FontUploader always updates the "current" font name to the last one loaded (such that it matches FontPreview's font of course).

### Location of Issue

FontUploader does not have a way of getting the google font name, so the issue starts there. App will need to send the name down to FontUploader once it receives the data from the GoogleFontUploader.

### Code Snippets

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mattysteves**
### Changes

- Added a useEffect Hook to FontUploader in order for the FontUploader component to update the font name of google fonts once they are loaded. The App component now stores the last uploaded google font's name for each side, and the FontUploader checks to see if it has changed using the Hook.

### Tests applied

- Tested loading of fonts from no previous font, then from file to google font, then google font to file. Seems to update accordingly for all combinations.

### Issue ticket number(s)

- #105 

### Checklist

- [ ] Documented
- [x] Linting tests successful
- [ ] merge updated prod branch into development branch

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -77,32 +84,64 @@ function App() {
...
   };
 
+  const setGoogleFontName = (fontData: string, side: string) => {
+    let fontName = ' ';
+    const parsedCss = postcss.parse(fontData);
+
+    // Find the first font-family value
+    parsedCss.walkAtRules('font-face', (atRule) => {
+      atRule.walkDecls('font-family', (decl) => {
+        fontName = decl.value.replace(/^['"]|['"]$/g, '');
```
Comment (calebWei) [inline src/components/App.tsx]: Something about the regex compromises the reliability check, try:
`fontName = decl.value.replace(/^(?:['"])|(?:['"]$)/g, '');`

(?:) should group the regex together.

File: src/components/App.tsx [hunk 2]
```diff
@@ -19,12 +20,18 @@ function App() {
   // State for the google font declarations on the left
   const [googleFontLeft, setGoogleFontLeft] = useState<string | null>(null);
 
+  // State for updating the font drop area with Google font name on the left
+  const [googelFontNameLeft, setGoogleFontNameLeft] = useState<string>('');
```
Comment (calebWei) [inline src/components/App.tsx]: Typo could be causing issues here

Comment (calebWei) [reply src/components/App.tsx]: actually not too certain, but sonarcloud is not happy about it

File: src/components/App.tsx [hunk 3]
```diff
@@ -19,12 +20,18 @@ function App() {
...
+
   // State for the selected font on the right
   const [selectedFontRight, setSelectedFontRight] = useState<File | null>(null);
 
   // State for the google font declarations on the right
   const [googleFontRight, setGoogleFontRight] = useState<string | null>(null);
 
+  // State for updating the font drop area with Google font name on the right
+  const [googelFontNameRight, setGoogleFontNameRight] = useState<string>('');
```
Comment (calebWei) [inline src/components/App.tsx]: There is also a typo here

**Review by calebWei — APPROVED** [review]
There are two code smells, but other than that it's ready for merge. Good work!

**Review by SamharAeron — CHANGES_REQUESTED** [review]
All looks good! Just please fix the sonar triggered code smells before merge!

See:
image

The google typo is the likely cause.

**Review by calebWei — APPROVED** [review]

**Review by SamharAeron — APPROVED** [review]
LGTM thanks for fixing the issues.

**Pull request merged**